src/EventSubscriber/PartySubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\EventSubscriber;
+
+use App\Entity\Party;
+use App\Repository\CasePartyRelationRepository;
+use App\Service\PartyHelper;
+use Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface;
+use Doctrine\ORM\Events;
+use Doctrine\Persistence\Event\LifecycleEventArgs;
+
+class PartySubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var CasePartyRelationRepository
+     */
+    private $relationRepository;
+    /**
+     * @var PartyHelper
+     */
+    private $partyHelper;
+
+    public function __construct(CasePartyRelationRepository $relationRepository, PartyHelper $partyHelper)
+    {
+        $this->relationRepository = $relationRepository;
+        $this->partyHelper = $partyHelper;
+    }
+
+    public function preUpdate(LifecycleEventArgs $args)
+    {
+        $object = $args->getObject();
+        $changeSet = $args->getEntityChangeSet();
+        if ($object instanceof Party && array_key_exists('name', $changeSet)) {
+            $relations = $this->relationRepository->findBy(['party' => $object->getId()]);
+
+            foreach ($relations as $relation) {
+                $relevantComplainant = $this->partyHelper->getSortingRelevantComplainant($relation->getCase());
```

Comment:
Why “relevant”? `$sortingComplainant = $this->partyHelper->getSortingComplainant($relation->getCase());` may be better naming.

Reply:
Renamed variable `$sortingComplainant` and removed `Relevant` from function name.